torch/nn/modules/module.py

```diff
@@ -1111,16 +1111,17 @@ def _maybe_warn_non_full_backward_hook(self, inputs, result, grad_fn):
                               "some grad_input. Please use register_full_backward_hook to get the documented "
                               "behavior.")
 
-    def register_forward_pre_hook(self, hook: Callable[..., None]) -> RemovableHandle:
+    def register_forward_pre_hook(self, hook: Callable[..., None], with_kwargs: bool = False) -> RemovableHandle:
         r"""Registers a forward pre-hook on the module.
 
         The hook will be called every time before :func:`forward` is invoked.
         It should have the following signature::
 
-            hook(module, input) -> None or modified input
+            hook(module, input, with_kwargs=False) -> None or modified input
 
-        The input contains only the positional arguments given to the module.
-        Keyword arguments won't be passed to the hooks and only to the ``forward``.
+        If ``with_kwargs`` is ``False`` the input and output contains only the
```

Comment:
Please ignore the doc changes for now - I decided I would do this after the implementation is nailed down